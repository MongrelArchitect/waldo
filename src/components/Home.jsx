import React, { useEffect } from 'react';
import { NavLink } from 'react-router-dom';
import Characters from './Characters';

export default function Home({
  characters, resetCharacters, setTime, timer,
}) {
  useEffect(() => {
    setTime(0);
    clearInterval(timer);
    resetCharacters();
  }, []);

  return (
    <div className="home">
      <h1>Find &apos;Em</h1>
      <div className="instructions">
        <h2>How to play:</h2>
        <span>
          Your goal is to find several characters hidden within a chaotic scene,
          just like the classic puzzle book series
          {' '}
          <a
            target="_blank"
            href="https://en.wikipedia.org/wiki/Where%27s_Wally%3F"
            rel="noreferrer"
          >
            &quot;Where&apos;s Waldo.&quot;
          </a>
          {' '}
          Click on the image when you&apos;ve found one of the characters, and
          select them from the dropdown to confirm. Try to get the best time
          possible, and compare your score to others on the scoreboard. Will you
          be the fastest? Let&apos;s go!
        </span>
      </div>
      <Characters characters={characters} />
      <div className="char-names">
        {characters.map((character) => (
          <span key={character.name}>{character.name}</span>
        ))}
      </div>
      <NavLink to="/play" className="play-link">
        START
      </NavLink>
    </div>
  );
}
