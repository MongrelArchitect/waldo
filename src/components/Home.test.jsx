import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import Home from './Home';

describe('Home component', () => {
  const chars = [
    { name: 'char one', img: '#', found: false },
    { name: 'char two', img: '##', found: false },
    { name: 'char three', img: '###', found: true },
  ];

  it('matches snapshot', () => {
    const { container } = render(
      <BrowserRouter>
        <Home characters={chars} setTime={jest.fn()} setTimer={jest.fn()} />
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
