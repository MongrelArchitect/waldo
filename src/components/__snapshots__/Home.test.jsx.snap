// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component matches snapshot 1`] = `
<div>
  <div
    class="home"
  >
    <h1>
      Find 'Em
    </h1>
    <div
      class="instructions"
    >
      <h2>
        How to play:
      </h2>
      <span>
        Your goal is to find several characters hidden within a chaotic scene, just like the classic puzzle book series
         
        <a
          href="https://en.wikipedia.org/wiki/Where%27s_Wally%3F"
          rel="noreferrer"
          target="_blank"
        >
          "Where's Waldo."
        </a>
         
        Click on the image when you've found one of the characters, and select them from the dropdown to confirm. Try to get the best time possible, and compare your score to others on the scoreboard. Will you be the fastest? Let's go!
      </span>
    </div>
    <span
      class="characters"
      data-testid="characters"
    >
      <img
        alt="char one"
        class="not-found"
        src="#"
      />
      <img
        alt="char two"
        class="not-found"
        src="##"
      />
      <img
        alt="char three"
        class="found"
        src="###"
      />
    </span>
    <div
      class="char-names"
    >
      <span>
        char one
      </span>
      <span>
        char two
      </span>
      <span>
        char three
      </span>
    </div>
    <a
      class="play-link"
      href="/play"
    >
      START
    </a>
  </div>
</div>
`;
